### PR TITLE
Add dotnet format github action

### PR DESCRIPTION
**Pull request by casz**
fixes #2082

### CODE REVIEW

**Comment by casz** [comment]
This depends on #2077 to not flag line endings warning.

**Comment by casz** [comment]
If you prefer I can convert it to an separate azure pipeline stage.

**Comment by arturcic** [comment]
Makes sense to use this on PRs and only on GitHubAction, to validate PRs are well formatted.

**Comment by arturcic** [comment]
@casz can you fix the code review requested and I will merge

**Comment by casz** [comment]
@arturcic perhaps you forgot to submit the review? I am not seeing any 😓

**Comment by arturcic** [comment]
> @arturcic perhaps you forgot to submit the review? I am not seeing any 😓

You're right, submitted now

File: .github/workflows/format.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: DotNet
```
[the remark this answers is not in the document]
Comment (casz) [reply .github/workflows/format.yml]: What about
```suggestion
name: Validate PR
```
and below suggestion:

File: .github/workflows/format.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+on: [pull_request]
+
+jobs:
+  format:
+    runs-on: ubuntu-latest
+    name: Format
```
Comment (casz) [inline .github/workflows/format.yml]:
```suggestion
    name: DotNet Format
```

**Comment by arturcic** [comment]
thank you so much for your contributions 👍 @casz

**Comment by casz** [comment]
Thanks for accepting my contribution :)

**Pull request merged**